pkg/runs/runsDownload.go

```diff
@@ -29,24 +30,84 @@ func DownloadArtifacts(
 
 	var err error = nil
 	var runs []galasaapi.Run
-	var artifactPaths []string
 
 	runs, err = GetRunsFromRestApi(runName, 0, 0, timeService, apiServerUrl)
 	if err == nil {
-		for _, run := range runs {
-			runId := run.GetRunId()
-			artifactPaths, err = GetArtifactPathsFromRestApi(runId, apiServerUrl)
-			if err == nil {
-				for _, artifactPath := range artifactPaths {
-					var artifactData io.Reader
-					artifactData, err = GetFileFromRestApi(runId, strings.TrimPrefix(artifactPath, "/"), apiServerUrl)
-					if err == nil {
-						err = WriteArtifactToFileSystem(fileSystem, runName, artifactPath, artifactData, forceDownload)
+		if len(runs) > 1 {
+			// get list of runs that are reRuns - get list of runs that are reRuns of each other
+			// create a map of lists of reRuns - key is queued time, value is the run
+
+			reRunsByQueuedTime := createMapOfReRuns(runs)
+
+			for _, reRunsList := range reRunsByQueuedTime {
+				if err == nil {
+					for reRunIndex, reRun := range reRunsList {
+						if err == nil {
+							directoryName := nameArtifactDownloadDirectory(reRun, reRunIndex)
+							err = downloadArtifactsToDirectory(apiServerUrl, directoryName, reRun, fileSystem, forceDownload)
+						}
 					}
 				}
 			}
+
+		} else {
+			for _, run := range runs {
```

Comment:
Surely there is only one run ? So do we need the loop ?

Reply:
agreed